discount/src/main/java/com/kyriba/conference/discount/service/DiscountServiceImpl.java

```diff
@@ -0,0 +1,59 @@
+package com.kyriba.conference.discount.service;
+
+import com.kyriba.conference.discount.api.dto.DiscountExternalDto;
+import com.kyriba.conference.discount.api.dto.DiscountResponse;
+import com.kyriba.conference.discount.api.dto.DiscountUpdateParamsDto;
+import com.kyriba.conference.discount.dao.DiscountRepository;
+import com.kyriba.conference.discount.domain.DiscountEntity;
+import com.kyriba.conference.discount.domain.DiscountType;
+import com.kyriba.conference.discount.util.DiscountDataMapper;
+import org.springframework.stereotype.Service;
+
+import javax.transaction.Transactional;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+/**
+ * @author Igor Lizura
+ */
+@Service
+@Transactional
+public class DiscountServiceImpl implements DiscountService{
+    private final DiscountRepository discountRepository;
+
+    public DiscountServiceImpl(DiscountRepository discountRepository) {
```

Comment:
Could be simplified with AllArgsConstructor

Reply:
fixed